tabrepo/utils/pickle_utils.py

```diff
@@ -49,23 +48,62 @@ def fetch_all_pickles(dir_path: str | Path, suffix: str = ".pkl") -> list[Path]:
     return file_paths
 
 
-def load_all_pickles(dir_path: str | Path) -> list[Any]:
+def fetch_all_pickles_fast(
+    dir_path: str | Path, suffix: str = ".pkl"
+) -> dict[str, list[Path]]:
```

Comment:
Is there a reason this is returning a dict and not a list? It doesn't seem like the keys are ever used elsewhere in the code in the PR.

Reply:
Looking closer, I assume it was for fetching the task_metadata.